pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/PurgeMinionClusterIntegrationTest.java

```diff
@@ -96,23 +98,27 @@ public void setUp()
     TableConfig purgeDeltaPassedTableConfig = createOfflineTableConfig();
     purgeDeltaPassedTableConfig.setTaskConfig(getPurgeTaskConfig());
 
+    setTableName(PURGE_OLD_SEGMENTS_WITH_NEW_INDICES_TABLE);
+    TableConfig purgeOldSegmentsWithNewIndicesTableConfig = createOfflineTableConfig();
+    purgeOldSegmentsWithNewIndicesTableConfig.setTaskConfig(getPurgeTaskConfig());
+
     addTableConfig(purgeTableConfig);
     addTableConfig(purgeDeltaPassedTableConfig);
     addTableConfig(purgeDeltaNotPassedTableConfig);
+    addTableConfig(purgeOldSegmentsWithNewIndicesTableConfig);
 
     // Unpack the Avro files
     List<File> avroFiles = unpackAvroData(_tempDir);
 
-    // Create and upload segments
-    ClusterIntegrationTestUtils.buildSegmentsFromAvro(avroFiles, purgeTableConfig, schema, 0, _segmentDir1, _tarDir1);
```

Comment:
Why did we consolidate everything to same directory?

Reply:
That wasn't needed. The schema and table configs for all those three tables were the same, so were the segments that were built. So there's no need to build the segments three times.